lib/sass/calcite-web/patterns/_tabs.scss

```diff
@@ -4,173 +4,94 @@
+    border-bottom-color: $border-active;
   }
-  @if ($include-right-to-left) {
-    html[dir="rtl"] & {
-      float: right;
-      margin-right: 0;
-      margin-left: -1px;
+
+  &:not(:last-child) {
+    margin-right: 1.25rem;
+
+    @if ($include-right-to-left) {
+      html[dir="rtl"] & {
+        margin-right: initial;
+        margin-left: 1.25rem;
+      }
+    }
+
+    @media screen and (max-width: $small) {
+      margin-right: 0.75rem;
+      @if ($include-right-to-left) {
+        html[dir="rtl"] & {
+          margin-left: 0.75rem;
+        }
+      }
     }
   }
 }
 
-@mixin tab-contents() {
+@mixin tab-contents($border) {
   @include box-sizing(border-box);
-  border: 1px solid $lighter-gray;
-  margin-top: -1px;
+  border-top: 1px solid $border;
 }
 
 @mixin tab-section() {
   @include box-sizing(border-box);
-  background-color: $white;
   display: none;
-  padding: $baseline/2;
+
   &.is-active {
     display: block;
   }
-  pre {
-    margin: 0;
-  }
 }
 
-@mixin tabs-gray() {
-  .tab-title    {
-    background-color: $white;
-    &.is-active {
-      background-color: $off-white;
-      border-bottom: 1px solid $off-white;
-    }
-  }
-  .tab-section  {
-    background-color: $off-white;
+@if $include-tabs==true {
+  .tab-nav {
+    @include tab-nav();
   }
-  pre code {
-    padding: 0;
-    border: none;
-    background-color: transparent;
+  .tab-section {
+    @include tab-section();
   }
-}
+  $tabStyles: 
+  ''$darkest-gray $lightest-gray $light-gray $off-black $blue,
+  '-dark'$lightest-gray $darker-gray $gray $off-white $white;
 
-@mixin tabs-transparent() {
-  .tab-title    {
-    background-color: transparent;
-    border: 1px solid transparent;
-    border-bottom: 1px solid $lighter-gray;
-    &:hover {
-      border-bottom: 1px solid $lighter-gray;
-    }
-    &.is-active {
-      @include transition(backround, $transition);
-      background-color: transparent;
-      border: 1px solid $lighter-gray;
-      border-bottom: 1px solid $white;
-    }
-  }
-  .tab-contents {
-    border-bottom: none;
-    border-left: none;
-    border-right: none;
-  }
-  .tab-section  {
-    background-color: transparent;
-    padding-left: 0;
-    padding-right: 0;
-  }
-}
+  @each $tabStyle in $tabStyles {
+    $name: nth($tabStyle, 1);
+    $color: nth($tabStyle, 2);
+    $border: nth($tabStyle, 3);
+    $border-hover: nth($tabStyle, 4);
+    $color-active: nth($tabStyle, 5);
+    $border-active: nth($tabStyle, 6);
 
-@mixin tabs-translucent() {
-  .tab-title {
-    background-color: $transparent-white;
-    background-image: none;
-    border: none;
-    border-top: 2px solid $transparent-white;
-    color: $off-black;
-    margin-right: 2px;
-    margin-bottom: 3px;
-    transition: none;
-    &:hover, &:focus {
-      background-color: $opaque-white;
-      border-top-color: $blue;
-      background-image: none;
+    .tabs#{$name} .tab-title {
+      @include tab-title($color, $border-hover, $color-active, $border-active);
```

Comment:
This will create an extremely verbose set of generated CSS. We need to use the cascade here, ie. the `tab-title` receives the full set of styles, but the theme-specific receive a subset. Right now, this current code will produce this:

```

.tabs .tab-title {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  font-size: 0.875rem;
  line-height: 1.5;
  padding: 0.5rem 0;
  flex: 0 1 auto;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-bottom: 3px solid transparent;
  cursor: pointer;
  color: #2b2b2b;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
}

.tabs .tab-title:active, .tabs .tab-title:hover, .tabs .tab-title:focus {
  outline: none;
  text-decoration: none;
  color: #151515;
  border-bottom-color: #bfbfbf;
}

.tabs .tab-title.is-active {
  color: #151515;
  border-bottom-color: #007ac2;
}

.tabs .tab-title:not(:last-child) {
  margin-right: 1.25rem;
}

html[dir="rtl"] .tabs .tab-title:not(:last-child) {
  margin-right: initial;
  margin-left: 1.25rem;
}

@media screen and (max-width: 480px) {
  .tabs .tab-title:not(:last-child) {
    margin-right: 0.75rem;
  }

  html[dir="rtl"] .tabs .tab-title:not(:last-child) {
    margin-left: 0.75rem;
  }
}

.tabs .tab-contents {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  border-top: 1px solid #eaeaea;
}

.tabs-dark .tab-title {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  font-size: 0.875rem;
  line-height: 1.5;
  padding: 0.5rem 0;
  flex: 0 1 auto;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-bottom: 3px solid transparent;
  cursor: pointer;
  color: #eaeaea;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
}

.tabs-dark .tab-title:active, .tabs-dark .tab-title:hover, .tabs-dark .tab-title:focus {
  outline: none;
  text-decoration: none;
  color: #f3f3f4;
  border-bottom-color: #808080;
}
```

See how the entire stucture gets duplicated?